shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/EraMap.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | This module provides the mapping from conceptual rule names to concrete
+-- rules for the Shelley era.
+module Shelley.Spec.Ledger.STS.EraMap () where
+
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Shelley (ShelleyEra)
+import Shelley.Spec.Ledger.STS.Bbody (BBODY)
+import Shelley.Spec.Ledger.STS.Chain (CHAIN)
+import Shelley.Spec.Ledger.STS.Deleg (DELEG)
+import Shelley.Spec.Ledger.STS.Delegs (DELEGS)
+import Shelley.Spec.Ledger.STS.Delpl (DELPL)
+import Shelley.Spec.Ledger.STS.Epoch (EPOCH)
+import Shelley.Spec.Ledger.STS.Ledger (LEDGER)
+import Shelley.Spec.Ledger.STS.Ledgers (LEDGERS)
+import Shelley.Spec.Ledger.STS.Mir (MIR)
+import Shelley.Spec.Ledger.STS.NewEpoch (NEWEPOCH)
+import Shelley.Spec.Ledger.STS.Newpp (NEWPP)
+import Shelley.Spec.Ledger.STS.Ocert (OCERT)
+import Shelley.Spec.Ledger.STS.Overlay (OVERLAY)
+import Shelley.Spec.Ledger.STS.Pool (POOL)
+import Shelley.Spec.Ledger.STS.PoolReap (POOLREAP)
+import Shelley.Spec.Ledger.STS.Ppup (PPUP)
+import Shelley.Spec.Ledger.STS.Rupd (RUPD)
+import Shelley.Spec.Ledger.STS.Snap (SNAP)
+import Shelley.Spec.Ledger.STS.Tick (TICK, TICKF)
+import Shelley.Spec.Ledger.STS.Tickn (TICKN)
+import Shelley.Spec.Ledger.STS.Utxo (UTXO)
+import Shelley.Spec.Ledger.STS.Utxow (UTXOW)
+
+type instance Core.EraSTS "BBODY" (ShelleyEra c) = BBODY (ShelleyEra c)
```

Comment:
I would be nice if this were more inheritable/reusable, e.g., "use the mapping from this era, but override only these rules".

Reply:
Yes, definitely. And here one moves towards type-level maps etc. Is it worth it? Maybe I'll investigate...

Reply:
Yup, I was thinking about the same.